diff -ru adchpp_2.8.0_source.orig//swig/SConscript adchpp_2.8.0_source//swig/SConscript
--- adchpp_2.8.0_source.orig//swig/SConscript	2012-02-24 07:15:25.531177571 +0100
+++ adchpp_2.8.0_source//swig/SConscript	2012-02-24 08:33:33.992187579 +0100
@@ -78,10 +78,6 @@
 	env.Append(SWIGFLAGS=['-c++','-threads','-Wall','-python', '-O', '-modern', '-features', 'autodoc=1'])
 	env['SWIGOUTDIR'] = Dir(dev.get_build_path('bin'))
 
-	if env['arch'] != 'x86':
-		print 'Skipping Python module'
-		return
-
 	version = sys.version[0] + sys.version[2]
 	
 	if version not in ('25','26','27'):
