diff -ru adchpp_2.8.0_source.orig//SConstruct adchpp_2.8.0_source//SConstruct
--- adchpp_2.8.0_source.orig//SConstruct	2012-02-23 05:36:29.568476919 +0100
+++ adchpp_2.8.0_source//SConstruct	2012-02-23 05:38:21.349477159 +0100
@@ -242,6 +242,6 @@
 
 		guide_path = '#/docs/user_guide/'
 		env.Command(doc_path + 'user_guide/basic_guide.html', guide_path + 'basic_guide.txt', asciidoc)
-		env.Command(doc_path + 'user_guide/novice_guide.html', guide_path + 'novice_guide.txt', asciidoc)
+		env.Command(doc_path + 'user_guide/expert_guide.html', guide_path + 'expert_guide.txt', asciidoc)
 		env.Command(doc_path + 'user_guide/images', guide_path + 'images', Copy('$TARGET', '$SOURCE'))
 
